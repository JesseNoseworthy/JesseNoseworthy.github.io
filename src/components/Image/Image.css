@import '../../styles/variables.css';
@import '../../styles/mixins.css';

:local(.Image) {

}