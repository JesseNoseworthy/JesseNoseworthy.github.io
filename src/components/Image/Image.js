import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Img = styled.img`
  width: 100%;
`;

const Image = ({ className, src, alt }) => src && <Img src={src} alt={alt} className={className} />;

Image.propTypes = {
  image: PropTypes.shape({
    src: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired
  })
};

export default Image;