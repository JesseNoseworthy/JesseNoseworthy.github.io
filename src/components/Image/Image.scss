.hero {
  position: absolute;
  top: 0;
  left: 0;
  z-index: -1;
  height: auto;
  width: 100%;
}