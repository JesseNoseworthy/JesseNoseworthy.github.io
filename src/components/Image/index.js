import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Image.css';

const Image = ({
  className,
  src,
  alt,
  animationType,
  ariaHidden
}) => (
  <img
    data-aos={animationType}
    className={classNames(styles.Image, className)}
    src={src}
    alt={alt}
    aria-hidden={ariaHidden}
  />
);

Image.propTypes = {
  className: PropTypes.string,
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  animationType: PropTypes.string,
  ariaHidden: PropTypes.string
};

export default Image;