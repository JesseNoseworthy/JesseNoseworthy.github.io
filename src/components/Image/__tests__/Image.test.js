import React from 'react';
import renderer from 'react-test-renderer';
import Image from '../index';

describe('Image', () => {
  describe('Snapshot tests', () => {
    it('default state (empty)', () => {
      const component = renderer.create(
        <Image
          src={'my-img.jpg'}
          alt={'this is alt text'}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('populated state', () => {
      const component = renderer.create(
        <Image
          className={'my-icon'}
          src={'my-img.jpg'}
          alt={'this is alt text'}
          animationType={'fade'}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
