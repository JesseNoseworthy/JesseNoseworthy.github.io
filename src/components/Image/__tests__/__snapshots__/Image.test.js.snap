exports[`Image Snapshot tests default state (empty) 1`] = `
<img
  alt="this is alt text"
  className=""
  data-aos={undefined}
  src="my-img.jpg" />
`;

exports[`Image Snapshot tests populated state 1`] = `
<img
  alt="this is alt text"
  className="my-icon"
  data-aos="fade"
  src="my-img.jpg" />
`;
